lib/worker.js

```diff
@@ -102,7 +102,13 @@ Communication.on('JOB', function(job) {
   }
 
   job.Response = new Promise(function(resolve) {
-    const filePath = (eslintignoreDir) ? Path.relative(eslintignoreDir, params.filePath) : params.filePath
+    let filePath
+    if (eslintignoreDir) {
+      filePath = Path.relative(eslintignoreDir, params.filePath)
+      process.chdir(eslintignoreDir)
+    } else {
+      filePath = Path.basename(params.filePath)
```

Comment:
Wouldn't it be nice to move [this](https://github.com/AtomLinter/linter-eslint/blob/issue335/lib/worker.js#L69) here?